packages/app/src/cli/commands/app/execute.ts

```diff
@@ -1,29 +1,89 @@
-import {appFlags} from '../../flags.js'
-import AppUnlinkedCommand, {AppUnlinkedCommandOutput} from '../../utilities/app-unlinked-command.js'
-import {AppInterface} from '../../models/app/app.js'
+import {appFlags, bulkOperationFlags} from '../../flags.js'
+import AppLinkedCommand, {AppLinkedCommandOutput} from '../../utilities/app-linked-command.js'
+import {linkedAppContext} from '../../services/app-context.js'
+import {storeContext} from '../../services/store-context.js'
+import {runBulkOperationQuery} from '../../services/bulk-operation-run-query.js'
 import {globalFlags} from '@shopify/cli-kit/node/cli'
-import {renderSuccess} from '@shopify/cli-kit/node/ui'
+import {renderSuccess, renderInfo, renderWarning} from '@shopify/cli-kit/node/ui'
+import {outputContent, outputToken} from '@shopify/cli-kit/node/output'
 
-export default class Execute extends AppUnlinkedCommand {
-  static summary = 'Execute app operations.'
+export default class Execute extends AppLinkedCommand {
+  static summary = 'Execute bulk operations.'
 
-  static description = 'Execute app operations.'
+  static description = 'Execute bulk operations against the Shopify Admin API.'
 
   static hidden = true
 
   static flags = {
     ...globalFlags,
     ...appFlags,
+    ...bulkOperationFlags,
   }
 
-  async run(): Promise<AppUnlinkedCommandOutput> {
-    await this.parse(Execute)
+  async run(): Promise<AppLinkedCommandOutput> {
+    const {flags} = await this.parse(Execute)
 
-    renderSuccess({
-      headline: 'Execute command ran successfully!',
-      body: 'Placeholder command. Add execution logic here.',
+    const appContextResult = await linkedAppContext({
+      directory: flags.path,
+      clientId: flags['client-id'],
+      forceRelink: flags.reset,
+      userProvidedConfigName: flags.config,
     })
 
-    return {app: undefined as unknown as AppInterface}
+    const store = await storeContext({
+      appContextResult,
+      storeFqdn: flags.store,
+      forceReselectStore: flags.reset,
+    })
+
+    renderInfo({
```

Comment:
The pattern we follow in every command is that we get the context (app context, store context) in the command file, but then all the logic is in a `service` file, so this file is kept very simple.